aws-distro-opentelemetry-node-autoinstrumentation/src/exporter/otlp/aws/common/aws-authenticator.ts

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { diag } from '@opentelemetry/api';
+import { getNodeVersion } from '../../../../utils';
+let SignatureV4: any;
+let HttpRequest: any;
+let defaultProvider: any;
+let Sha256: any;
+
+let dependenciesLoaded = false;
+
+if (getNodeVersion() >= 16) {
+  try {
+    defaultProvider = require('@aws-sdk/credential-provider-node').defaultProvider;
+    Sha256 = require('@aws-crypto/sha256-js').Sha256;
+    SignatureV4 = require('@smithy/signature-v4').SignatureV4;
+    HttpRequest = require('@smithy/protocol-http').HttpRequest;
+    dependenciesLoaded = true;
+  } catch (error) {
+    diag.error(`Failed to load required AWS dependency for SigV4 Signing: ${error}`);
+  }
+} else {
+  diag.error('SigV4 signing requires at least Node major version 16');
+}
+
+// See: https://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-authenticating-requests.html
+export const AUTHORIZATION_HEADER = 'authorization';
+export const X_AMZ_DATE_HEADER = 'x-amz-date';
+export const X_AMZ_SECURITY_TOKEN_HEADER = 'x-amz-security-token';
+export const X_AMZ_CONTENT_SHA256_HEADER = 'x-amz-content-sha256';
+
+export class AwsAuthenticator {
+  private endpoint: URL;
+  private region: string;
+  private service: string;
+
+  constructor(endpoint: string, service: string) {
+    this.endpoint = new URL(endpoint);
```

Comment:
nit - add comment to mention that we pre-validate the endpoint such that `new URL()` will not throw.